core/app/models/spree/store_credit.rb

```diff
@@ -184,7 +189,17 @@ def update_amount(amount, reason, user_performing_update)
     self.action = ADJUSTMENT_ACTION
     self.update_reason = reason
     self.action_originator = user_performing_update
-    save
+    if save
```

Comment:
I still don't think this is quite what I'm looking for. I would prefer something like:

```Ruby
if action_amount > 0
  credit(action_amount, user_performing_update)
else
  debit(action_amount, user_performing_update)
else
```

and move to a system where that is what's happening. Then the credit and debit methods are the ones which alter the ledger.